README.md

```diff
@@ -1,16 +1,695 @@
+|-------|--------------------------|--------------------------------------------------------|----------|---------|
+| `v`   | `SemVer`                 | SemVer of the UCAN invocation this schema (v0.1.0)     | Yes      |         |
+| `prf` | `[&UCAN]`                | UCANs that supply the authority to run this invocation | Yes      |         |
+| `run` | `"*" or {String : Task}` | Which UCAN capabilities to run                         | Yes      |         |
+| `nnc` | `String`                 | A unique nonce, to distinguish each invocation         | Yes      |         |
+| `ext` | `Any`                    | Non-normative extended fields                          | No       | `null`  |
+    
+### 3.1.1 Version
+
+The `v` field MUST contain the version of the invocation object  schema.
+
+### 3.1.2 Proofs
+
+The `prf` field MUST contain CIDs pointing to the UCANs that provide the authority to run these tasks. The elements of this array MUST be sorted in ascending numeric order. Restricting the outmost UCANs to only the authority required for the current invocation is RECOMMENDED.
+
+### 3.1.3 Run Capabilities
+
+The `run` field MUST reference the tasks contained in the UCAN are to be run during the invocation. To run all tasks in the underlying UCAN, the `"*"` value MUST be used. If only specific tasks (or [pipelines](#5-promise-pipelining)) are intended to be run, then they MUST be prefixed with an arbitrary label and treated as a UCAN attenuation: all tasks MUST be backed by a matching capability of equal or greater authority.
+
+#### 3.1.3.1 Promises
+
+The only difference from general capabilities is that [promises](#5-promise-pipelining) MAY also be used as inputs to attenuated fields.
+
+If a capability input has the key `"_"` and the value is a promise, the input MUST be discarded and only used for determining sequencing tasks.
+
+### 3.1.4 Nonce
+
+The `nnc` field MUST contain a unique nonce. This field exists to enable making the CID of each invocation unique. While this field MAY be an empty string, it is NOT RECOMMENDED.
+
+### 3.1.5 Extended Fields
+
+The OPTIONAL `ext` field MAY contain arbitrary data. If not present, the `ext` field MUST be interpreted as `null`, including for [signature](#315-signature).
+
+## 3.2 IPLD Schema
+
+``` ipldsch
+type SignedInvocation struct {
+  inv Invocation (rename "ucan/invoke") 
+  sig VarSig
+}
+
+type Invocation struct {
+  v   SemVer  -- Version
+  prf [&UCAN] -- Authority to run this invocation
+  run Scope   -- Which tasks to invoke
+  nnc String  -- Nonce
+  ext nullable Any (implicit null) -- Extended fields
+}
+
+type Scope union {
+  | All "*"
+  | {String : Task}
+}
+
+type Task struct {
+  with   URI 
+  do     Ability
+  inputs Any
+  meta   {String : Any}
+} 
+```
+
+## 3.3 JSON Examples
+
+### 3.3.1 Run All
+
+ ``` js
+{
+  "ucan/invoke": {
+    "v": "0.1.0",
+    "prf": [
+      {"/": "bafkreie2cyfsaqv5jjy2gadr7mmupmearkvcg7llybfdd7b6fvzzmhazuy"},
```

Comment:
It might really help to have illustration of what those CIDs point to.

Reply:
Yes, @zeeshanlakhani also gave similar feedback. I'm going to add a bunch of annotations inline and likely an `/examples` directory that has step-by-step explications.